### PR TITLE
Module File Storing Completed

### LINKED ISSUE #160: Module Files Update Cover Image

### PR DESCRIPTION
**Pull request by dmols**
Module Data is now stored in its own folder within the courseID folder. This way the cover image of the course will be easily seen as different from the files stored for the modules. Added a symbol checker, preventing any non-alphanumeric filenames. This may be a little annoying for the user but it'll help prevent the code from being vulnerable to harmful commands.

### CODE REVIEW

File: frontend/toolkit-webapp/src/pages/ModuleCreator.js [hunk 1]
```diff
@@ -134,7 +134,20 @@ function ModuleCreator(props) {
...
+                const pdfTypePath = module.pdf.name.split('.')
+
+                // Grabbing the actual filename minus extension so that we can validate alphanumeric inputs
+                var val = pdfTypePath[pdfTypePath.length - 2];
+                var RegEx = /[^0-9a-z]/i;
+                var isValid = !(RegEx.test(val));
+
+                // Input contains non-alphanumeric values so we must alert the user to rename the file 
+                if (isValid === false) {
+                    alert('Invalid file type. Please upload an image for which name is alphanumeric and has no spaces.')
```
Comment (mbrancato31) [inline frontend/toolkit-webapp/src/pages/ModuleCreator.js]: should it be PDF? it says image

File: frontend/toolkit-webapp/src/pages/ModuleCreator.js [hunk 2]
```diff
@@ -169,7 +182,20 @@ function ModuleCreator(props) {
...
+                const fileTypePath = module.file.name.split('.')
+
+                // Grabbing the actual filename minus extension so that we can validate alphanumeric inputs
+                var val = fileTypePath[fileTypePath.length - 2];
+                var RegEx = /[^0-9a-z]/i;
+                var isValid = !(RegEx.test(val));
+
+                // Input contains non-alphanumeric values so we must alert the user to rename the file 
+                if (isValid === false) {
+                    alert('Invalid file type. Please upload an image for which name is alphanumeric and has no spaces.')
```
Comment (mbrancato31) [inline frontend/toolkit-webapp/src/pages/ModuleCreator.js]: here too, should be a file right?

File: frontend/toolkit-webapp/src/pages/ModuleCreator.js [hunk 3]
```diff
@@ -205,7 +231,20 @@ function ModuleCreator(props) {
...
+
+                // Grabbing the actual filename minus extension so that we can validate alphanumeric inputs
+                var val = videoTypePath[videoTypePath.length - 2];
+
+                var RegEx = /[^0-9a-z]/i;
+                var isValid = !(RegEx.test(val));
+
+                // Input contains non-alphanumeric values so we must alert the user to rename the file 
+                if (isValid === false) {
+                    alert('Invalid file type. Please upload an image for which name is alphanumeric and has no spaces.')
```
Comment (mbrancato31) [inline frontend/toolkit-webapp/src/pages/ModuleCreator.js]: video

File: frontend/toolkit-webapp/src/pages/ModuleEditor.js [hunk 4]
```diff
@@ -137,6 +137,20 @@ function ModuleEditor(props) {
...
+                const pdfTypePath = module.pdf.name.split('.')
+
+                // Grabbing the actual filename minus extension so that we can validate alphanumeric inputs
+                var val = pdfTypePath[pdfTypePath.length - 2];
+                var RegEx = /[^0-9a-z]/i;
+                var isValid = !(RegEx.test(val));
+
+                // Input contains non-alphanumeric values so we must alert the user to rename the file 
+                if (isValid === false) {
+                    alert('Invalid file type. Please upload an image for which name is alphanumeric and has no spaces.')
```
Comment (mbrancato31) [inline frontend/toolkit-webapp/src/pages/ModuleEditor.js]: pdf

File: frontend/toolkit-webapp/src/pages/ModuleEditor.js [hunk 5]
```diff
@@ -172,7 +186,20 @@ function ModuleEditor(props) {
...
+                const fileTypePath = module.file.name.split('.')
+
+                // Grabbing the actual filename minus extension so that we can validate alphanumeric inputs
+                var val = fileTypePath[fileTypePath.length - 2];
+                var RegEx = /[^0-9a-z]/i;
+                var isValid = !(RegEx.test(val));
+
+                // Input contains non-alphanumeric values so we must alert the user to rename the file 
+                if (isValid === false) {
+                    alert('Invalid file type. Please upload an image for which name is alphanumeric and has no spaces.')
```
Comment (mbrancato31) [inline frontend/toolkit-webapp/src/pages/ModuleEditor.js]: file

File: frontend/toolkit-webapp/src/pages/ModuleEditor.js [hunk 6]
```diff
@@ -209,7 +236,18 @@ function ModuleEditor(props) {
...
-                // handle image
+                const videoTypePath = module.video.name.split('.')
+                var val = videoTypePath[videoTypePath.length - 2];
+
+                var RegEx = /[^0-9a-z]/i;
+                var isValid = !(RegEx.test(val));
+
+                // Input contains non-alphanumeric values so we must alert the user to rename the file 
+                if (isValid === false) {
+                    alert('Invalid file type. Please upload an image for which name is alphanumeric and has no spaces.')
```
Comment (mbrancato31) [inline frontend/toolkit-webapp/src/pages/ModuleEditor.js]: video

**Review by mbrancato31 — CHANGES_REQUESTED** [review]
everything looks good, only some minor changes in the alerts

**Review by collyntheshots — APPROVED** [review]
Beyond what has been commented on everything looks good

**Review by nlyra — APPROVED** [review]
There are a lot of good fixes in this branch that we will need, but there is still an issue with text modules where if you try to edit a text module it will crash due to some undefined variables.

**Pull request merged**